### PR TITLE
PT-12036: Add capture order operation

### PR DESCRIPTION
**Pull request by EugeneOkhriemnko**
### Jira-link:

https://virtocommerce.atlassian.net/browse/PT-12036

### CODE REVIEW

**Review by OlegoO — DISMISSED** [review]

File: src/VirtoCommerce.OrdersModule.Data/Model/CaptureEntity.cs [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+        public virtual Capture ToModel(Capture model)
+        {
+            return (Capture)ToModel((OrderOperation)model);
+        }
+
+        public override OrderOperation ToModel(OrderOperation operation)
+        {
+            if (operation is not Capture capture)
+            {
+                throw new ArgumentException($"{nameof(operation)} argument must be of type {nameof(Refund)}", nameof(operation));
```
Comment (ksavosteev) [inline src/VirtoCommerce.OrdersModule.Data/Model/CaptureEntity.cs:48]: {nameof(Capture)}

File: src/VirtoCommerce.OrdersModule.Data/Model/CaptureEntity.cs [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+        public virtual void Patch(CaptureEntity target)
+        {
+            Patch((OperationEntity)target);
+        }
+
+        public override void Patch(OperationEntity target)
+        {
+            if (target is not CaptureEntity capture)
+            {
+                throw new ArgumentException($"{nameof(target)} argument must be of type {nameof(RefundEntity)}", nameof(target));
```
Comment (ksavosteev) [inline src/VirtoCommerce.OrdersModule.Data/Model/CaptureEntity.cs:121]: {nameof(CaptureEntity)}

File: src/VirtoCommerce.OrdersModule.Web/Scripts/blades/capture-add.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
+angular.module('virtoCommerce.orderModule')
+    .controller('virtoCommerce.orderModule.captureAddController', [
+        '$scope',
+        'platformWebApp.bladeNavigationService',
+        'virtoCommerce.orderModule.order_res_customerOrders',
+    function ($scope, bladeNavigationService, customerOrders) {
+        var blade = $scope.blade;
+        blade.title = 'orders.blades.refund-add.title';
```
Comment (ksavosteev) [inline src/VirtoCommerce.OrdersModule.Web/Scripts/blades/capture-add.js:8]: 'orders.blades.capture-add.title'

File: src/VirtoCommerce.OrdersModule.Web/Scripts/blades/capture-add.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+                    bladeNavigationService.setError(data.errorMessage, blade);
+                }
+            }, function (error) {
+                blade.isLoading = false;
+                bladeNavigationService.setError('Error ' + error.status, blade);
+            })
+        };
+
+        $scope.getRefundReasons = function () {
+            return refundReasonsService.getRefundReasons();
```
Comment (ksavosteev) [inline src/VirtoCommerce.OrdersModule.Web/Scripts/blades/capture-add.js:54]: captureReasons

**Review by ksavosteev — DISMISSED** [review]

**Review by ksavosteev — APPROVED** [review]

**Review by ksavosteev — APPROVED** [review]

**Pull request merged**